React Implementation - Sprint 2

OliOliKT opened the pull request.

OliOliKT committed:
Add font styles and remove unnecessary .git files

emhberg committed:
IndividualPost done, and minor changes to footer and topbar

I created the IndividualPost page, and then I also created UserCommentsComponent on the same branch... upsi.. I created an initial array for the comments (can be changed later). Other than that I made some minor changes to footer and header, so they work optimal now.

emhberg committed:
Merge pull request #5 from OliOliKT/SingularPost

IndividualPost done, and minor changes to footer and topbar

BuurPrime committed:
Few additions

jdisitu committed:
Add UserSettingsSidebar component

OliOliKT committed:
Make PostStats, PostFilter components and make MyPosts page

jdisitu committed:
Update sidebar styling and structure

jdisitu committed:
Add new styling to navigation sidebar

OliOliKT committed:
Add User Settings to MyPosts

emhberg committed:
Changed small things to the UserCommentComponent.

Added a border to the UserCommentComponent, so it it will be possible to like a comment in the future.
Minor changes was also done to the topBar, as it acted a bit strange. So now it is always on top of the other components.

emhberg committed:
Merge pull request #6 from OliOliKT/SingularPost

Changed small things to the UserCommentComponent.

jdisitu committed:
add entire user settings page with all its components

jdisitu committed:
Add UserSettings page

OliOliKT committed:
Small modifications to a lot of components and pages

OliOliKT committed:
Merge pull request #8 from OliOliKT/modificationsForSprint1

Small modifications to a lot of components and pages

OliOliKT committed:
Refactor all files

OliOliKT committed:
Merge pull request #9 from OliOliKT/modificationsForSprint1

Refactor all files

OliOliKT committed:
Remove unnecessary files

OliOliKT committed:
Merge pull request #10 from OliOliKT/modificationsForSprint1

Remove unnecessary files

BuurPrime committed:
Added routing and dropdown menu in topbar component

- Changed topbarcomponent alot (both style and functionality)
- The Feed Navigation works, but does not change color accordingly
- Noticed that something is off on the pages in terms of spacing

emhberg committed:
Initial Log in page created

Initial Log in page created

BuurPrime committed:
Merge branch 'development' of https://github.com/OliOliKT/StealthStories into development

OliOliKT committed:
Change MyPosts so that it uses center instead of margin

OliOliKT committed:
Merge pull request #11 from OliOliKT/testBranchOliver

Change MyPosts so that it uses center instead of margin

BuurPrime committed:
Fixed footer and Trending block

emhberg committed:
Database for User created

Imported parse library and tested it out for user. Some navigation has also be changed, so it is up to date.

emhberg committed:
Merge pull request #12 from OliOliKT/TryingDataBaseForUser

Database for User created

OliOliKT committed:
Connect feed to database

OliOliKT committed:
Merge pull request #13 from OliOliKT/testBranchOliver

Connect feed to database

emhberg committed:
Creating a new user

Things added to create a new user, I also had to add a logout function, so changes have been done to the topbar as well.

emhberg committed:
Merge pull request #14 from OliOliKT/CreateUserPage

Creating a new user

jdisitu committed:
Added post functioanlity for liking and writing comments on posts

BuurPrime committed:
tried fixing the navigation

BuurPrime committed:
Merge branch 'discoverFeed' into development

OliOliKT committed:
Add post functionality via database

BuurPrime committed:
Fized trending box (again...) and fixed h1 margin issue

BuurPrime committed:
Fixed feedNavigation

OliOliKT committed:
Merge pull request #15 from OliOliKT/testBranchOliver

Add post functionality via database

emhberg committed:
Merge pull request #16 from OliOliKT/PostFunctionality

Added post functioanlity for liking and writing comments on posts

BuurPrime committed:
Small changes to individualpost

Commit:
add comment button redirection to individual page

emhberg committed:
created error messages for new user and for logging in

BuurPrime committed:
Merge pull request #17 from OliOliKT/CreateAlertForWrongPassword

created error messages for new user and for logging in

emhberg committed:
Merge pull request #18 from OliOliKT/PostFunctionality

add comment button redirection to individual page

Commit:
fixed bug where comment would lead to undefined URL. Add logo to Individual posts

BuurPrime committed:
Fixed spacing between posts

Commit:
add functionality to enter button

emhberg committed:
The current user in user settings

The current user is now displayed in the user settings together with its ID. I deleted the "edit" button as well, as it was wrong that a user could change its email and ID

emhberg committed:
Merge pull request #19 from OliOliKT/GettingCurrentUserInUserSettings

The current user in user settings

Commit:
fixed merge conflicts

Commit:
fixed issue where shift enter also posted

jdisitu committed:
Merge pull request #20 from OliOliKT/PostFunctionality

Post functionality

OliOliKT committed:
Connect likes to database

OliOliKT committed:
Fix merge conflict

jdisitu committed:
Merge pull request #21 from OliOliKT/sprint2BranchOliver

Sprint2 branch oliver

emhberg committed:
Added gaps to posts

emhberg committed:
Merge pull request #22 from OliOliKT/FixingPosts

Added gaps to posts

BuurPrime committed:
Made comment section

mircealungu commented on src/components/AccountInformation.js:
```diff
@@ -0,0 +1,56 @@
+          {showUserId ? (
+            <span>{currentUser.id}</span>
+          ) : (
+            <span>*************</span>
```
why are we showing stars when the uid is not defined?

jdisitu replied:
We have a hide button which toggles between the user id and the stars.

mircealungu commented on src/components/AddCommentButton.js:
```diff
@@ -0,0 +1,12 @@
+function AddCommentButton() {
+  return (
+    <div className="add-comment-button">
+        <p className="add-comment-text"> + Add comment </p>
```
a component should be reusable. 
This is just a hardcoded button with the text that's not changeable.
why is i a component then?

jdisitu replied:
resolved

mircealungu commented on src/components/ButtonAndText.css:
```diff
@@ -1,25 +1,22 @@
+  color: hsl(152, 93%, 65%);
+}
+
+.iconLink {
```
kebab-case?

jdisitu replied:
on it

mircealungu commented on src/components/CommentComponent.js:
```diff
@@ -1,49 +1,68 @@
 import "./CommentComponent.css";
 
-const WriteComment = () => {
+const WriteComment = ({ postId, closeCommentModal, onCommentPosted }) => {
```
You never name a component FooComponent. It's clearly a component based on it's structure.

jdisitu replied:
on it

mircealungu commented on src/components/CommentButtonAndText.js:
```diff
@@ -1,13 +1,15 @@
   return (
-    <div className="iconAndText">
-      <a href="#" className="iconLink">
+    <div className="iconAndText" onClick={onComment}>
```
name all event handlers with `handle` maybe? as you do in the next file?

jdisitu replied:
on it

mircealungu commented on src/components/CommentComponent.js:
```diff
@@ -1,49 +1,68 @@
+  };
+
   return (
     <section className="writeComment">
```
based on the code we see here there's no need for this component to exist. simply move the section inside of the CommentBox. OR/?

mircealungu commented on src/components/CreateUserButtonAndText.js:
```diff
@@ -0,0 +1,73 @@
+import { useState } from "react";
```
looks good.

mircealungu commented on src/components/FormInput.js:
```diff
@@ -0,0 +1,66 @@
+import { useNavigate } from "react-router-dom";
+import { Link } from "react-router-dom";
+
+function LogInButtonAndFormInput() {
```
it's good practice to have same name for component and container file. (https://github.com/itu-tid/lecture-notes/blob/main/Tech-TopUps/5-Coding-Practices.md#component-and-file-in-which-it-is-defined-should-have-the-same-name)

jdisitu replied:
on it

mircealungu commented on src/components/FormInput.js:
```diff
@@ -0,0 +1,66 @@
+      navigate("/DiscoverFeed");
+    } catch (error) {
+      console.error("Error while logging in user", error);
+      setError("You typed in the wrong password!");
```
your error message might be misleading at times. do you know why?

jdisitu replied:
on it

mircealungu commented on src/components/Post.js:
```diff
@@ -1,46 +1,69 @@
+  const [updatedSipCount, setUpdatedSipCount] = useState(sipCount); // State to store updated sipCount
+  const navigate = useNavigate();
+
+
```
these many empty lines tell me that you're not using format-on-save with Prettier (https://github.com/itu-tid/lecture-notes/blob/main/Tech-TopUps/5-Coding-Practices.md#install-an-automatic-formatter-and-configure-it-to-format-on-save)

jdisitu replied:
on it

mircealungu commented on src/components/Post.js:
```diff
@@ -1,46 +1,69 @@
+  const handleSip = async () => {
+    try {
+      const Post = Parse.Object.extend("Post");
+      const query = new Parse.Query(Post);
```
i'm not sure i understand the 'sip' behavior. based on what i see here it alternates between 0 and 1?

mircealungu commented on src/components/TwoFactorAuthentication.js:
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import './TwoFactorAuthentication.css';
+
+const TwoFactorAuthentication = () => {
```
:) i don't think you'll have time to implement that...

mircealungu commented on src/components/writePost.js:
```diff
@@ -0,0 +1,142 @@
+  useEffect(() => {
+    async function fetchUserData() {
+      try {
+        const currentUser = Parse.User.current();
```
this is a very roundabout way of getting the username.
why do you have to get the session? 
you have the user id in the `Parser.User.current()`. You can simply query the User table based on that. 
or am I missing something?

BuurPrime committed:
Merge branch 'development' of https://github.com/OliOliKT/StealthStories into development

mircealungu reviewed (changes requested):
overall very good work!
please have a look at the comments and address the ones you can and agree with. 

mircea and @igawaclawska

Commit:
added functioanlity for comments

BuurPrime committed:
Merge pull request #24 from OliOliKT/commentFunctionality

added functioanlity for comments

emhberg committed:
Hower Fixed on posts

Hower Fixed on posts

emhberg committed:
Merge pull request #25 from OliOliKT/FixHoverButtons

Hower Fixed on posts

BuurPrime committed:
Fixed fednavigation hover bug, changed spacing between posts

OliOliKT committed:
Fix footer spacing, Fix My Posts showing own posts

OliOliKT committed:
Merge pull request #26 from OliOliKT/sprint3BranchOliver

Fix footer spacing, Fix My Posts showing own posts

OliOliKT committed:
Add comment logic

OliOliKT committed:
Merge pull request #27 from OliOliKT/sprint3BranchOliver

Add comment logic

BuurPrime committed:
Update writePost.js

BuurPrime committed:
Merge branch 'development' into NotificationFeature

emhberg committed:
back button on create new user

Created back button for create new user

emhberg committed:
Merge pull request #28 from OliOliKT/BackToLoginButton

back button on create new user

BuurPrime committed:
Live addition of comments, setup user settings page in grid

BuurPrime committed:
Merge branch 'NotificationFeature' into development

BuurPrime committed:
Post will now be posted live (although will some delay for whatever reason)

BuurPrime committed:
Comments now re-size based on their content

BuurPrime committed:
Fixed layout issues

BuurPrime committed:
Footer Fix

BuurPrime committed:
Fixed more footer issues

OliOliKT committed:
Add mood filter functionality

OliOliKT committed:
Merge pull request #29 from OliOliKT/moodFilter

Add mood filter functionality

OliOliKT committed:
Small changes

OliOliKT committed:
Merge pull request #30 from OliOliKT/moodFilter

Small changes

OliOliKT committed:
Change filters

OliOliKT committed:
Merge pull request #31 from OliOliKT/moodFilter

Change filters

emhberg committed:
Fixed username for create user

The username works now when creating a new user

emhberg committed:
Merge pull request #32 from OliOliKT/CreatingUserName

Fixed username for create user

emhberg committed:
Updated sidebar

Updated the sidebar to have the pages we are going to create and linked to the relevant pages

emhberg committed:
Merge pull request #33 from OliOliKT/ChangeSideBar

Updated sidebar

emhberg committed:
Created AboutUs page

emhberg committed:
Merge pull request #34 from OliOliKT/AboutUsPage

Created AboutUs page

emhberg committed:
Created Privacy and Safety page

emhberg committed:
Merge pull request #35 from OliOliKT/SafetyAndPrivacyPage

Created Privacy and Safety page

BuurPrime committed:
Changes in post functionality to make it load faster

BuurPrime committed:
Code formatting

OliOliKT committed:
Add popularity filter, add real data to MyPosts stats

OliOliKT committed:
Merge pull request #36 from OliOliKT/moodFilter

Add popularity filter, add real data to MyPosts stats

BuurPrime committed:
Formatting and cleaning up

BuurPrime committed:
Merge branch 'development' of https://github.com/OliOliKT/StealthStories into development

BuurPrime committed:
Fixed comment user icon size bug

BuurPrime committed:
comment

BuurPrime committed:
Merge branch 'DatabaseRestructuring' into development

BuurPrime committed:
Fixed slow loading of posts on feed

BuurPrime committed:
Fixed comment hover issue

BuurPrime committed:
Added hover to add comment button

BuurPrime committed:
About us page updated

BuurPrime committed:
Updated privacy and safety page

emhberg committed:
Added kebab-case

Added kebab-case to create user and login pages

emhberg committed:
Merge pull request #37 from OliOliKT/ChangeToKebabLogin

Added kebab-case

OliOliKT committed:
Add grey box to not implemented features and add pointers to buttons

OliOliKT committed:
Merge pull request #38 from OliOliKT/makeStuffGrey

Add grey box to not implemented features and add pointers to buttons

OliOliKT committed:
Merge branch 'master' into development
